src/MQTTClient.c

```diff
@@ -2551,7 +2551,7 @@ static MQTTPacket* MQTTClient_cycle(int* sock, ELAPSED_TIME_TYPE timeout, int* r
 				*rc = MQTTProtocol_handlePubrecs(pack, *sock);
 			}
 			else if (pack->header.bits.type == PUBREL)
-				*rc = MQTTProtocol_handlePubrels(pack, *sock);
+				*rc = MQTTProtocol_handlePubrels(pack, *sock, 0);
```

Comment:
The MQTTClient interface expects the caller of MQTTClient_receive to free the topicName and message, so this function is to not delete the topicName.